src/main/resources/api/data-repository-openapi.yaml

```diff
@@ -6711,6 +6762,63 @@ components:
           items:
             type: string
 
+    SnapshotBuilderAccessRequest:
+      type: object
+      properties:
+        name:
+          type: string
+        researchPurposeStatement:
+          type: string
+        datasetRequest:
+          $ref: '#/components/schemas/SnapshotBuilderRequest'
+      description: Request for access to a snapshot.
+
+    SnapshotBuilderRequest:
+      type: object
+      properties:
+        cohorts:
+          type: array
+          items:
+            $ref: '#/components/schemas/SnapshotBuilderCohort'
+        conceptSets:
+          type: array
+          items:
+            $ref: '#/components/schemas/SnapshotBuilderDatasetConceptSets'
+        valueSets:
+          type: array
+          items:
+            $ref: '#/components/schemas/SnapshotBuilderFeatureValueGroup'
+      description: Specification of requested a snapshot, returned in SnapshotBuilderAccessRequest.
+
+    SnapshotBuilderCohort:
+      type: object
+      properties:
+        name:
+          type: string
+        criteriaGroups:
+          type: array
+          items:
+            $ref: '#/components/schemas/SnapshotBuilderCriteriaGroup'
+      description: Selection of participants.
+
+    SnapshotBuilderCriteriaGroup:
+      type: object
+      properties:
+        name:
+          type: string
+        criteria:
+          type: array
+          items:
+            oneOf:
+              - $ref: '#/components/schemas/SnapshotBuilderDomainOption'
+              - $ref: '#/components/schemas/SnapshotBuilderProgramDataOption'
```

Comment:
`ProgramDataOption` might work fine for both cases, even though it is representing a different thing. It may be that the same data is needed for both.

Reply:
I could also do SnapshotBuilderConcept (which is the root of DomainOption) and ProgramDataOption, just to keep them separate.